File: app/src/main/java/jp/bellware/echo/filter/GainDetector.java
package jp.bellware.echo.filter;

/**
 * ゲイン取得
 */
public class GainDetector {
    private float max = 0;

    public void filter(float s) {
        if (s < 0)
            s *= -1;
        if (s > max) {
            max = s;
        }
    }

    public void reset() {
        max = 0;
    }

    public float getMax() {
        return max;
    }

}

File: localDataStore/src/main/java/jp/bellware/echo/util/filter/GainDetector.kt
package jp.bellware.echo.util.filter

/**
 * ゲイン取得
 */
class GainDetector {
    /**
     * ゲイン
     */
    var max = 0f
        private set

    /**
     * サンプルを追加する
     */
    fun add(sample: Float) {
        var s = sample
        if (s < 0)
            s *= -1f
        if (s > max) {
            max = s
        }
    }

    /**
     * リセットする
     */
    fun reset() {
        max = 0f
    }
}
